etna/metrics/base.py

```diff
@@ -322,7 +345,7 @@ def __call__(self, y_true: TSDataset, y_pred: TSDataset) -> Union[float, Dict[st
 
         segments = df_true.columns.get_level_values("segment").unique()
 
-        metrics_per_segment: Dict[str, float]
+        metrics_per_segment: Dict[str, Optional[float]]
```

Comment:
We should also probably do smth with NaNs returned from metric functions:
- We should cast them into `None` inside `__call__`
- We should cast `NaNs` into `None` inside the functional metrics

Reply:
Probably yes, but
- It makes implementation of functional metric more difficult
- It doesn't save us from getting `NaNs` from functional metrics

Reply:
* There will be more logic there, for example, if you use `nanmean`, an error may occur, it should be handled as well.
* And when can `nan` occur there? I proceeded from the fact that `None` is returned if everything is empty.
* This can also be processed, if you convert it to the `object` type, you can put `None` there, but most likely we will lose a little in performance in such a case.

Reply:
- I saw only `RuntimeWarning: Mean of empty slice`. What error are you talking about?
- There are two scenarios: we allow `NaNs` to occur, we have some error that makes them to occur...
- We could probably just return a `list`. I'm worried that returning `list` or `ndarray[object]` isn't that people regularly expect from functions like `mean_squared_error`, etc.

Reply:
If we are going to move this logic into functional metric we should probably update `ArrayLike` or make a separate type hint for return value.

Reply:
* Yes, you're right. Sorry for misleading I referred to this
* If `nan` is a result of some kind of error, it still would be converted to `None` at the last step in the functional metric, wouldn't it ?
* This is true. Should we consider not replacing `nan` with `None`?